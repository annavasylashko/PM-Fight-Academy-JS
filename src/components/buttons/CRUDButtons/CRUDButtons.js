import React from "react";

import styles from "../../MainStyles/mainStyles.module.css";

const CRUDButtons = (props) => {
  return (
    <div className={styles.btnContainer}>
      {props.isCreateButtonVisible && (
        <button
          type="submit"
          disabled={props.isCreateButtonDisabled}
          className={styles.addBtn}
        >
          <p className={styles.addBtnText}>Done</p>
        </button>
      )}
      {props.isUpdateButtonVisible && (
        <button
          onClick={props.updateHandler}
          disabled={props.isUpdateButtonDisabling}
          className={styles.addBtn}
        >
          <p className={styles.addBtnText}>Update</p>
        </button>
      )}

      {props.isDeleteButtonVisible && (
        <button
          onClick={props.deleteHandler}
          disabled={props.isDeleteButtonDisabled}
          className={styles.addBtn}
        >
          <p className={styles.addBtnText}>Delete</p>
        </button>
      )}
    </div>
  );
};

export default CRUDButtons;
