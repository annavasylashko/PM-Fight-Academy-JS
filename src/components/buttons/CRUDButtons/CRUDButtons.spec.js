import { render } from '@testing-library/react';
import CRUDButtons from './CRUDButtons';

describe('CRUDButtons', () => {
  it('should show all buttons', () => {
    const props = {
      isCreateButtonVisible: true,
      isUpdateButtonVisible: true,
      isDeleteButtonVisible: true,
    };
    const { container } = render(<CRUDButtons {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should show no buttons', () => {
    const props = {
      isCreateButtonVisible: false,
      isUpdateButtonVisible: false,
      isDeleteButtonVisible: false,
    };
    const { container } = render(<CRUDButtons {...props} />);

    expect(container).toMatchSnapshot();
  });
});
