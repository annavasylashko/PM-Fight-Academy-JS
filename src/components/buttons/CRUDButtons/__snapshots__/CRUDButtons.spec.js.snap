// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CRUDButtons should show all buttons 1`] = `
<div>
  <div
    class="btnContainer"
  >
    <button
      class="addBtn"
      data-testid="crud-button-submit"
      type="submit"
    >
      <p
        class="addBtnText"
      >
        Done
      </p>
    </button>
    <button
      class="addBtn"
      data-testid="crud-button-update"
    >
      <p
        class="addBtnText"
      >
        Update
      </p>
    </button>
    <button
      class="addBtn"
      data-testid="crud-button-delete"
    >
      <p
        class="addBtnText"
      >
        Delete
      </p>
    </button>
  </div>
</div>
`;

exports[`CRUDButtons should show no buttons 1`] = `
<div>
  <div
    class="btnContainer"
  />
</div>
`;
